e2e-app/e2e/src/typeahead/typeahead.po.ts

```diff
@@ -0,0 +1,34 @@
+import {$, Key} from 'protractor';
+import {sendKey} from '../tools';
+
+export class TypeaheadPage {
+
+  getInputBefore() {
+    return $('#first');
+  }
+
+  getInput() {
+    return $('#typeahead');
+  }
+
+  getDropdown() {
+    return $('ngb-typeahead-window');
+  }
+
+  getDropdownItems() {
+    return this.getDropdown().$$('button');
+  }
+
+  async clearText() {
+    await this.getInput().clear();
+  }
+
+  async getText() {
+    return this.getInput().getAttribute('value');
+  }
+
+  async reset() {
```

Comment:
In datepicker I also added expectations in `reset()` just to be sure that future modifications wont break start conditions:
- both inputs text is empty
- typeahead is closed
- body is focused

Otherwise we should reload URL on each test